### ISSUE TITLE
Remove deprecated threadable.whenThreaded

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4550]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4550|
|Type|task|
|Created|2010-07-06 00:29:30Z|
|Branch|https://github.com/twisted/twisted/tree/remove-threadable-whenThreaded-4550|

```whenThreaded``` in [twisted/python/threadable.py](twisted/python/threadable.py) was deprecated in r14762, time to remove it.

<details><summary>Searchable metadata</summary>

```
trac-id__4550 4550
type__task task
reporter__thijs thijs
priority__lowest lowest
milestone__ 
branch__branches_remove_threadable_whenThreaded_4550 branches/remove-threadable-whenThreaded-4550
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1278376170000000 1278376170000000
changetime__1288126014000000 1288126014000000
version__None None
owner__ 
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

(In [30163]) Branching to 'remove-threadable-whenThreaded-4550'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

(In [30164]) Removed twistd.python.threadable.whenThreaded.
Refs [#4550](https://github.com/twisted/twisted/issues/4550)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli removed owner|
|-|-|

Removed whenThreaded.  There were no tests for it or other references.  Also bumped copyright.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli commented|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-threadable-whenThreaded-4550)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess set owner to @cyli|
|-|-|

Thanks for working on this, cyli. As with [#4551](https://github.com/twisted/twisted/issues/4551), the NEWS file should use the removal construction described in ReviewProcess#Newsfiles.

Beyond the removal, it looks like "test-case-name: twisted.python.threadable" should actually be twisted.test.test_threadable, if you want to change that while you're touching the file.

Other than that, please merge!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

(In [30175]) Changed news file to conform to http://twistedmatrix.com/trac/wiki/ReviewProcess and changed the threadable.py test to be test_threadable.

Refs [#4550](https://github.com/twisted/twisted/issues/4550)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli commented|
|-|-|

Done and done.  Thanks for the review!  Merging.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_9"></a>|@cyli set status to `closed`|
|-|-|

(In [30176]) Merge remove-threadable-whenThreaded-4550: Remove deprecated threadable.whenThreaded


Author: cyli

Reviewer: jesstess

Fixes: [#4550](https://github.com/twisted/twisted/issues/4550)


twisted.python.threadable.whenThreaded, deprecated since 2005, has been removed.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
